### PR TITLE
Feature/509 player UI refactor

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
## 🧰 Issue
**Convert All PlayerUi, define props, optimize code and move all logic functions in to `./helpers` folder (only usingsections, some sections may be from old Mapping component)**

sections using state:
- [x] packages/client/src/Components/Channel.jsx
- [x] packages/client/src/Components/MessageCreatorChatChannel.jsx
- [x] packages/client/src/Components/MessageListItem.jsx
- [x] packages/client/src/Components/Modals/InsightsModal.jsx
- [x] packages/client/src/Components/Modals/ModalSwitch/ModalSwitchPlayer.jsx
- [x] packages/client/src/Store/PlayerUi.jsx
- [x] packages/client/src/Views/AdminAndInsightsTabsContainer.jsx
- [x]  packages/client/src/Views/ChannelTabsContainer/ChannelTabsContainer.jsx
- [x]  packages/client/src/Views/GameAdmin.jsx
- [x]  packages/client/src/Views/GameChannels.jsx
- [x]  packages/client/src/Views/InsightsChannel.jsx
- [x]  packages/client/src/Views/MessagesListChatChannel.jsx
- [x]  packages/client/src/Views/PlayerUi.jsx
- [x]  remove - packages/client/src/Components/GameControls.jsx  (used from @serge/components)
- [x]  packages/client/src/Components/MessageCreator.jsx 
- [x]  remove - packages/client/src/Components/MessagePreviewPlayerUi.jsx (not used)
- [x]  remove - packages/client/src/Components/TurnProgression.jsx (used from @serge/components)
- [x]  add SelectedForce object to player ui state and remove "Repeating find"s
- [x] fix double time "json" usage

other sections:
 
- [x] packages/client/src/Components/LoaderScreen.jsx
  - [x] packages/client/src/Components/Loader.jsx
- [x] packages/client/src/Views/PlayerUiInitiate.jsx
- [x] packages/client/src/Views/PlayerUiLandingScreen.jsx
- [x] packages/client/src/Views/PlayerUiLobby.jsx
  - [x] packages/client/src/Components/Inputs/TextInput.jsx (Changed to use TexInput from @serge/components)
- [x] packages/client/src/Views/GameChannelsWithTour.jsx

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

File: packages/client/src/Components/Channel.jsx [hunk 1]
```diff
@@ -9,20 +9,20 @@ import {
   openMessage,
   markAllAsRead,
 } from "../ActionsAndReducers/playerUi/playerUi_ActionCreators";
-import { PlayerStateContext } from "../Store/PlayerUi";
+import { usePlayerUiState, usePlayerUiDispatch } from "../Store/PlayerUi";
 
 import "@serge/themes/App.scss";
 
 class Channel extends Component {
```
Comment (lilitkarapetyan) [inline packages/client/src/Components/Channel.jsx]: Need to change to use React.FC

File: packages/client/src/Components/GameControls.jsx [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import React, { Component } from "react";
 import { nextGameTurn } from "../ActionsAndReducers/playerUi/playerUi_ActionCreators";
-import { PlayerStateContext } from "../Store/PlayerUi";
+import { usePlayerUiState, usePlayerUiDispatch } from "../Store/PlayerUi";
 
 import "@serge/themes/App.scss";
 
 class GameControls extends Component {
```
Comment (lilitkarapetyan) [inline packages/client/src/Components/GameControls.jsx]: Need to change to use React.FC

**Comment by IanMayo** [comment]
Minor note.  When we're logged in as a player, we can click on the Role text at the bottom right, which opens a URL that contains the game & pass-code.

This isn't working in this branch, the pass-code gets rejected.  I _Think_ the issue is because `allForces` isn't available in `usePlayerUiState` when opening the new URL in `PlayerUI.tsx`.
Screen Shot 2021-01-05 at 13 24 41

**Comment by IanMayo** [comment]
@lilitkarapetyan - I've fixed some DeepScan issues, but 3-4 remain. Would you mind fixing them as part of your updates please?

**Comment by IanMayo** [comment]
@lilitkarapetyan  - I think I've fixed the ts-ignore entries you highlighted. There are lots more in `handle world state changes`  but I'm happy to fix those in a separate issue/PR.  #577

**Comment by IanMayo** [comment]
@lilitkarapetyan - does that mean this PR is ready for review/merge?

**Review by IanMayo — APPROVED** [review]

**Pull request merged**